### PR TITLE
Impose ordering on generated docs/code elements, to ensure consistency

### PR DESCRIPTION
**Pull request by petemoore**
Note, commit 2882a4b63654b0fe96010eb37fe0495cc2fefa05 contains a lot of noise since it is the result of the code generation process, which has imposed an ordering for the first time. Therefore you only need to review the changes from the other commits.

For example when locally reviewing, you could revert 2882a4b63654b0fe96010eb37fe0495cc2fefa05 and then compare to current master (fe4fd8bbd8e7d0a4dfec4e23c88100a143396a05).

### CODE REVIEW

File: bin/update-apis.js [hunk 1]
```diff
@@ -52,7 +58,10 @@ var updateDocs = function() {
       "//  - " + api.reference.baseUrl,
       "var " + instanceName(name) + " = new taskcluster." + name + "(options);",
       "```"
-    ].concat(api.reference.entries.filter(function(entry) {
+    ].concat(_.sortBy(api.reference.entries, function(entry) {
+      // Sort by function name: http://bugzil.la/1200519
+      return entry.name;
```
Comment (jonasfj) [inline bin/update-apis.js]: These are ordered...

File: bin/update-apis.js [hunk 2]
```diff
@@ -85,7 +95,10 @@ var updateDocs = function() {
       "//  - " + api.reference.exchangePrefix,
       "var " + instanceName(name) + " = new taskcluster." + name + "(options);",
       "```"
-    ].concat(api.reference.entries.filter(function(entry) {
+    ].concat(_.sortBy(api.reference.entries, function(entry) {
+      // Sort by entry name: http://bugzil.la/1200519
+	  return entry.name;
```
Comment (jonasfj) [inline bin/update-apis.js]: They are ordered by declaration...

Comment (petemoore) [inline bin/update-apis.js]: Quite right - my mistake.

File: bin/update-apis.js (hunk 1 above)
Comment (petemoore) [inline bin/update-apis.js]: Indeed - my mistake.

**Comment by jonasfj** [comment]
@petemoore, I like this but don't break circle ci, this is one of the few repos where it supposed to work :)

File: circle.yml [hunk 3]
```diff
@@ -1,3 +1,10 @@
...
-    version: 0.10.37
\ No newline at end of file
+    version: 0.10.37
+
+test:
+  post:
+    - bin/update-apis.js update
+    - git status # debug to see why the following command fails, if it does
+    - git diff | cat   # moar debug, hopefully will help
+    - /bin/bash -c 'test $(git status --porcelain | wc -l) == 0' # check that generated files have been committed
```
Comment (jonasfj) [inline circle.yml]: I get why this is nice... but it's not necessary to test for...
If so should probably not be in post test...

IMO, it's not worth the investment yet... maybe when the reference format is more solid. Let's not over-invest at this stage we have bigger fish to fry.

**Comment by jonasfj** [comment]
I like this... stable json makes a lot of sense...

**Comment by petemoore** [comment]
I've squashed the commits as they were way too many - now there is just a single commit which contains everything, and the circleci tests are fixed. :)

**Pull request merged**